src/App.tsx

```diff
@@ -1,29 +1,59 @@
-import { Component } from 'react';
-import './App.scss';
+/* eslint-disable no-console */
+import { useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
+import { getMovie } from './api/movieApi';
+import './App.scss';
 
-interface State {
-  movies: Movie[];
-}
+export const App: React.FC = () => {
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [foundedMovie, setFoundedMovie] = useState<Movie | null>(null);
+  const [IsMovieFounded, setIsMovieFounded] = useState(true);
 
-export class App extends Component<{}, State> {
-  state: State = {
-    movies: [],
+  const searchMovieTitle = async (title:string) => {
+    try {
+      const movie = await getMovie(title);
+
+      if (movie.Response === 'True') {
+        setFoundedMovie(movie);
+        setIsMovieFounded(true);
+      } else {
+        setIsMovieFounded(false);
+      }
+    } catch (error) {
+      console.error(
+        'An error has occurred',
+      );
+    }
   };
 
-  render() {
-    const { movies } = this.state;
+  const addToList = () => {
+    if (!foundedMovie) {
+      return;
+    }
+
+    if (movies.length) {
+      if (movies.find(movie => movie.imdbID === foundedMovie.imdbID)) {
+        return;
+      }
+    }
```

Comment:
Unnecessary nesting)

`movies.length && movies.find(movie => movie.imdbID === foundedMovie.imdbID)`

Reply:
The default value is `[]` so find will anyways return undefined

```suggestion
      if (movies.find(movie => movie.imdbID === foundedMovie.imdbID)) {
        return;
      }
```